renter/programbuilder.go

```diff
@@ -0,0 +1,192 @@
+package renter
+
+import (
+	"bytes"
+	"encoding/binary"
+	"errors"
+
+	"go.sia.tech/core/net/rhp"
+	"go.sia.tech/core/types"
+)
+
+// A ProgramBuilder constructs programs for the renter to execute on a host.
+type ProgramBuilder struct {
+	instructions                           []rhp.Instruction
+	requiresFinalization, requiresContract bool
```

Comment:
I would put these on separate lines so that the padding on other fields isn't so extreme